x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/expression/function/scalar/datetime/DateTrunc.java

```diff
@@ -6,98 +6,205 @@
 import static org.elasticsearch.xpack.sql.expression.function.scalar.datetime.BinaryDateTimeProcessor.BinaryDateOperation.TRUNC;
 
 public class DateTrunc extends BinaryDateTimeFunction {
 
+    private static final long SECONDS_PER_MINUTE = 60;
+    private static final long SECONDS_PER_HOUR = SECONDS_PER_MINUTE * 60;
+    private static final long SECONDS_PER_DAY = SECONDS_PER_HOUR * 24;
+
     public enum Part implements DateTimeField {
 
-        MILLENNIUM(dt -> {
-            int year = dt.getYear();
-            int firstYearOfMillenium = year - (year % 1000);
-            return dt
-                .with(ChronoField.YEAR, firstYearOfMillenium)
-                .with(ChronoField.MONTH_OF_YEAR, 1)
-                .with(ChronoField.DAY_OF_MONTH, 1)
-                .toLocalDate().atStartOfDay(dt.getZone());
-            },"millennia"),
-        CENTURY(dt -> {
-            int year = dt.getYear();
-            int firstYearOfCentury = year - (year % 100);
-            return dt
-                .with(ChronoField.YEAR, firstYearOfCentury)
-                .with(ChronoField.MONTH_OF_YEAR, 1)
-                .with(ChronoField.DAY_OF_MONTH, 1)
-                .toLocalDate().atStartOfDay(dt.getZone());
-            }, "centuries"),
-        DECADE(dt -> {
-            int year = dt.getYear();
-            int firstYearOfDecade = year - (year % 10);
-            return dt
-                .with(ChronoField.YEAR, firstYearOfDecade)
-                .with(ChronoField.MONTH_OF_YEAR, 1)
-                .with(ChronoField.DAY_OF_MONTH, 1)
-                .toLocalDate().atStartOfDay(dt.getZone());
-            }, "decades"),
-        YEAR(dt -> dt
-            .with(ChronoField.MONTH_OF_YEAR, 1)
-            .with(ChronoField.DAY_OF_MONTH, 1)
-            .toLocalDate().atStartOfDay(dt.getZone()),
-            "years", "yy", "yyyy"),
-        QUARTER(dt -> {
-            int month = dt.getMonthValue();
-            int firstMonthOfQuarter = (((month - 1) / 3) * 3) + 1;
-            return dt
-                .with(ChronoField.MONTH_OF_YEAR, firstMonthOfQuarter)
-                .with(ChronoField.DAY_OF_MONTH, 1)
-                .toLocalDate().atStartOfDay(dt.getZone());
-            }, "quarters", "qq", "q"),
-        MONTH(dt -> dt
-                .with(ChronoField.DAY_OF_MONTH, 1)
-                .toLocalDate().atStartOfDay(dt.getZone()),
-            "months", "mm", "m"),
-        WEEK(dt -> dt
-                .with(ChronoField.DAY_OF_WEEK, 1)
-                .toLocalDate().atStartOfDay(dt.getZone()),
-            "weeks", "wk", "ww"),
-        DAY(dt -> dt.toLocalDate().atStartOfDay(dt.getZone()), "days", "dd", "d"),
-        HOUR(dt -> {
-            int hour = dt.getHour();
-            return dt.toLocalDate().atStartOfDay(dt.getZone())
-                .with(ChronoField.HOUR_OF_DAY, hour);
+        MILLENNIUM((Object r) -> {
+            if(r instanceof ZonedDateTime){
```

Comment:
Here and elsewhere, please fix the formatting by 

- adding a space after `if` and before `(`
- adding a space after `)` and before `{`
- add a space after `,` in method arguments